### ISSUE TITLE
[WARNING] HypothesisDeprecationWarning: `HealthCheck.all()` is deprecated

### ISSUE THREAD

**Issue by kgutwin** [schemathesis/schemathesis#1739]
**Checklist**

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)

**Describe the bug**
Running either the `st` command line tool or in my own pytest suite, I receive a deprecation warning:

```
.../venv/lib/python3.10/site-packages/hypothesis/_settings.py:467: HypothesisDeprecationWarning: `Healthcheck.all()` is deprecated; use `list(HealthCheck)` instead.
    The `hypothesis codemod` command-line tool can automatically refactor your code to fix this warning.
  note_deprecation(
```

The warning isn't reproduced on all endpoints; however, as shown below, it is reported at least on the POST operation on https://example.schemathesis.io/api/payload .

**To Reproduce**

1. Run `st run --endpoint /api/payload https://example.schemathesis.io/openapi.json`
2. Output:

```
================================= Schemathesis test session starts =================================
Schema location: https://example.schemathesis.io/openapi.json
Base URL: https://example.schemathesis.io/api
Specification version: Open API 3.0.2
Workers: 1
Collected API operations: 1

/Users/kgutwin/bitbucket/nih-nimhd/schare-pigeon/project-pigeon/venv/lib/python3.10/site-packages/hypothesis/_settings.py:467: HypothesisDeprecationWarning: `Healthcheck.all()` is deprecated; use `list(HealthCheck)` instead.
    The `hypothesis codemod` command-line tool can automatically refactor your code to fix this warning.
  note_deprecation(
POST /api/payload .                                                                           [100%]

============================================= SUMMARY ==============================================

Performed checks:
    not_a_server_error                    101 / 101 passed          PASSED 

Hint: You can visualize test results in Schemathesis.io by using `--report` in your CLI command.

======================================== 1 passed in 12.76s ========================================
```

**Expected behavior**
No warnings to be shown.

**Environment (please complete the following information):**

- OS: MacOS 12.6.3
- Python version: 3.10.10
- Schemathesis version: 3.19.1
- Hypothesis version: 6.75.3
- Spec version: Open API 3.0.2
